### PR TITLE
🌱 Remove e2e templates from the code base

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
Remove e2e templates from the codebase (those files now are generated on the fly before running tests)

This is a follow up of https://github.com/kubernetes-sigs/cluster-api/pull/3651#discussion_r489406752

### CODE REVIEW

**Comment by fabriziopandini** [comment]
@vincepri apidiff is failing, but there is nothing that triggers in `scripts/ci-apidiff.sh` 😞

File: test/e2e/data/infrastructure-docker/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+*.yaml
```
[the remark this answers is not in the document]
Comment (fabriziopandini) [reply test/e2e/data/infrastructure-docker/.gitignore]: done

**Comment by vincepri** [comment]
Is there something that's modifying the tree before running apidiff?

**Comment by vincepri** [comment]
/hold

This breaks apidiff

**Comment by sedefsavas** [comment]
Adding` *-template*.yaml `to the `.gitignore `makes this fail. Not sure why.

**Comment by sedefsavas** [comment]
FYI: apidiff doesn't break when `test/e2e/data/infrastructure-docker/*-template` is added to the main .gitignore.

**Comment by fabriziopandini** [comment]
API diff passed, thanks @sedefsavas, really appreciated!

**Review by vincepri — APPROVED** [review]
/approve
/hold cancel
/assign @sedefsavas

**Pull request merged**